experimental/DrinfeldHeckeAlgebras/src/DrinfeldHeckeAlgebra.jl

```diff
@@ -0,0 +1,669 @@
+  end
+end
+
+function (A::DrinfeldHeckeAlgebra)(a)
+  A(group_algebra(A)(a))
+  try
+    return A(group_algebra(A)(a))
+  catch 
+    throw(ArgumentError("Element cannot be cast into the given Drinfeld-Hecke algebra")) 
+  end
+end
+
+#######################################
+# Generic Drinfeld-Hecke algebra generation
+#######################################
+
+@doc raw"""
+    generic_drinfeld_hecke_algebra(G::MatrixGroup{T}, R::Ring=base_ring(G)) where {T <: FieldElem}
+
+Create a generic (parametrized) Drinfeld-Hecke algebra over the ring ```R``` for the group ```G```.
+
+# Examples
+```jldoctest
+julia> G = matrix_group(matrix(QQ, [-1 0;0 -1]))
+Matrix group of degree 2
+  over rational field
+
+julia> generic_drinfeld_hecke_algebra(G)
+Drinfeld-Hecke algebra
+   for Matrix group of degree 2 over QQ
+with generators
+   x1, x2, g1
+
+defined by Drinfeld-Hecke form over base ring
+   Multivariate polynomial ring in 2 variables over QQ
+with parameters 
+   t1, t2
+given by alternating bilinear forms
+   [1   0] => [  0   t1]
+   [0   1]    [-t1    0]
+
+   [-1    0] => [  0   t2]
+   [ 0   -1]    [-t2    0]
+```
+"""
+function generic_drinfeld_hecke_algebra(G::MatrixGroup{T}, R::Ring=base_ring(G)) where {T <: FieldElem}
+  κ = generic_drinfeld_hecke_form(G,R)
+
+  return DrinfeldHeckeAlgebra(κ)
+end
+
+#######################################
+# Evaluate parameters of generic Drinfeld-Hecke algebra
+#######################################
+
+@doc raw"""
+    evaluate_parameters(A::DrinfeldHeckeAlgebra, values::Vector)
+
+Evaluate the parameters of the Drinfeld-Hecke algebra ```A``` by the values given in ```values```.
+
+# Examples
+```jldoctest
+julia> G = matrix_group(matrix(QQ, [-1 0;0 -1]))
+Matrix group of degree 2
+  over rational field
+
+julia> A = generic_drinfeld_hecke_algebra(G)
+Drinfeld-Hecke algebra
+   for Matrix group of degree 2 over QQ
+with generators
+   x1, x2, g1
+
+defined by Drinfeld-Hecke form over base ring
+   Multivariate polynomial ring in 2 variables over QQ
+with parameters 
+   t1, t2
+given by alternating bilinear forms
+   [1   0] => [  0   t1]
+   [0   1]    [-t1    0]
+
+   [-1    0] => [  0   t2]
+   [ 0   -1]    [-t2    0]
+
+julia> evaluate_parameters(A, [1,2])
+Drinfeld-Hecke algebra
+   for Matrix group of degree 2 over QQ
+with generators
+   x1, x2, g1
+
+defined by Drinfeld-Hecke form over base ring
+   Multivariate polynomial ring in 2 variables over QQ
+with parameters 
+   t1, t2
+given by alternating bilinear forms
+   [1   0] => [ 0   1]
+   [0   1]    [-1   0]
+
+   [-1    0] => [ 0   2]
+   [ 0   -1]    [-2   0]
+```
+"""
+function evaluate_parameters(A::DrinfeldHeckeAlgebra, values::Vector)
+  G = group(A)
+  R = base_ring(A)
+  
+  if !(R isa MPolyRing)
+    throw(ArgumentError("The given form does not have any parameters."))
+  end
+
+  # If A is zero, there is nothing to do
+  if is_zero(form(A)) return drinfeld_hecke_algebra(G,R) end
```

Comment:
More idiomatic would be:
```suggestion
  is_zero(form(A)) && return drinfeld_hecke_algebra(G,R)
```

Reply:
Should this really be before the check for `length(values) == n` ? That check might uncover invalid inputs after all?